### PR TITLE
Week 7

### PR DESCRIPTION
**Pull request by siddhantdalvi3**

### CODE REVIEW

**Comment by siddhantdalvi3** [comment]
Please ignore W6 files in this PR

File: Week-6/W6Exercise6.3/Index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const sortArr = (arr) => {
```
Comment (kapilggg10) [inline Week-6/W6Exercise6.3/Index.js:1]: can you think of a batter approach?

File: Week-6/W6Exercise6.5/Index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const findPairWithDifference = (arr, difference) => {
```
Comment (kapilggg10) [inline Week-6/W6Exercise6.5/Index.js:1]: any other approach?

File: Week-6/W6Exercise6.6/Index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const threeSumClosest = (nums, target) => {
+    let closest = 0;
+    let minDiff = Infinity;
+    nums.sort((a, b) => a - b);
+    for (let i = 0; i < nums.length - 2; i++) {
```
Comment (kapilggg10) [inline Week-6/W6Exercise6.6/Index.js:5]: Explain this for loop in laymen terms if you have some 5 mins?

File: Week-7/Readme.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
+# Pesto Week 7 assignments
```
Comment (kapilggg10) [inline Week-7/Readme.md:1]: A great thanks for this Readme.md file always.

File: Week-7/Helper/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+class SLLNode {
+  data;
+  next;
+  constructor(value) {
+    this.data = value;
+    this.next = null;
+  }
+}
+const createSLL = (arr) => {
```
Comment (kapilggg10) [inline Week-7/Helper/index.js:9]: good one 👍

File: Week-7/W7Exercise7.3/Index.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+const { createSLL } = require("../Helper");
```
Comment (kapilggg10) [inline Week-7/W7Exercise7.3/Index.js:1]: Good use of Helper File 👍

File: Week-7/W7Exercise7.3/Index.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    fast = fast.next.next;
+    if (slow === fast) {
+      return true;
+    }
+  }
+  return false;
+};
+
+// create a loop in a linked list from last node to xth node from first node.
+const createLoop = (head, x) => {
```
Comment (kapilggg10) [inline Week-7/W7Exercise7.3/Index.js:18]: awesome 👍

File: Week-7/W7Exercise7.4/Index.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    exp.length === 0 ||
+    !exp.includes("(") ||
+    !exp.includes("[") ||
+    !exp.includes("{")
+  )
+    return false;
+
+  let stack = [];
+
+  for (let i = 0; i < exp.length; i++) {
```
Comment (kapilggg10) [inline Week-7/W7Exercise7.4/Index.js:18]: Explain this in laymen terms

File: Week-7/W7Exercise7.4/Index.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    if (exp[i] === "(" || exp[i] === "[" || exp[i] === "{") {
+      stack.push(exp[i]);
+    } else if (exp[i] === ")" || exp[i] === "]" || exp[i] === "}") {
+      if (stack.length === 0) {
+        return false;
+      }
+      let top = stack.pop();
+      if (
```
Comment (kapilggg10) [inline Week-7/W7Exercise7.4/Index.js:26]: ok 👍

File: Week-7/W7Exercise7.5/Index.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+const findNextGreaterElement = (arr) => {
+  let _arr = [];
+  let found = false;
+  for (let i = 0; i < arr.length - 1; i++) {
+    if (arr[i] < arr[i + 1]) {
+      _arr.push(arr[i + 1]);
+    } else {
+      for (let j = i; j < arr.length; j++) {
```
Comment (kapilggg10) [inline Week-7/W7Exercise7.5/Index.js:8]: are you sure, time complexity for this is O(N)?

**Review by kapilggg10 — APPROVED** [review]
Going good, Try to add some laymen terms comments for loops and function.

**Pull request merged**